kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -358,6 +358,41 @@ impl<'tcx> GotocHook<'tcx> for MemCmp {
     }
 }
 
+/// This hook intercepts calls to `std::ptr::align_offset<T>` as CBMC's memory model has no concept
+/// of alignment of allocations, so we would have to non-deterministically choose an alignment of
+/// the base pointer, add the pointer's offset to it, and then do the math that is done in
+/// `library/core/src/ptr/mod.rs`. Instead, we choose to always return `usize::MAX`, per
+/// `align_offset`'s documentation, which states: "It is permissible for the implementation to
+/// always return usize::MAX. Only your algorithm’s performance can depend on getting a usable
+/// offset here, not its correctness."
+pub struct AlignOffset;
+
+impl<'tcx> GotocHook<'tcx> for AlignOffset {
+    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+        let name = with_no_trimmed_paths!(tcx.def_path_str(instance.def_id()));
+        name == "std::ptr::align_offset"
```

Comment:
I think we should include `core::ptr::align_offset` as well.

Reply:
Actually, they are two different methods. the `std` one is safe, while the `core` one is unsafe. Instead of using the name here, I would recommend using the `align_offset` lang_item instead, since it is way more reliable.

In this case, you would be overriding the `core` version, which is unsafe and it is UB to execute it with an input that is not power of 2. So please add a check for that.